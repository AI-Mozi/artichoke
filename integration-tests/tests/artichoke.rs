use integration_tests::run;

const BINARY: &str = "artichoke";
const FIXTURES_ROOT: &str = "./tests/fixtures/";

#[cfg(target_family = "windows")]
#[test]
fn help_windows() {
    insta::assert_toml_snapshot!(run(BINARY, &["--help"]).unwrap());
}

#[cfg(target_family = "unix")]
#[test]
fn help_unix() {
    insta::assert_toml_snapshot!(run(BINARY, &["--help"]).unwrap());
}

#[test]
fn hello_world() {
    let app_name = "hello_world.rb";
    let path = format!("{}{}", FIXTURES_ROOT, app_name);
    insta::assert_toml_snapshot!(run(BINARY, &[&path]).unwrap());
}

#[test]
fn fizz_buzz() {
    let app_name = "fizz_buzz.rb";
    let path = format!("{}{}", FIXTURES_ROOT, app_name);
    insta::assert_toml_snapshot!(run(BINARY, &[&path]).unwrap());
}
