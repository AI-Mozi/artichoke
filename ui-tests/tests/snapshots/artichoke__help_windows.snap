---
source: tests/artichoke.rs
assertion_line: 9
expression: "run(BINARY, &[\"--help\"]).unwrap()"

---
call_args = ['--help']
status = 0
stdout = '''
"artichoke 0.1.0-pre.0"
"Artichoke is a Ruby made with Rust."
""
"USAGE:"
"    artichoke.exe [OPTIONS] [ARGS]"
""
"ARGS:"
"    <programfile>     "
"    <arguments>...    "
""
"OPTIONS:"
"        --copyright                 print the copyright"
"    -e <commands>                   one line of script. Several -e\'s allowed. Omit [programfile]"
"    -h, --help                      Print help information"
"    -V, --version                   Print version information"
"        --with-fixture <fixture>    file whose contents will be read into the `$fixture` global"'''
stderr = ''
